src/js/models/documentexports.js

```diff
@@ -48,6 +49,35 @@ define(function (require, exports, module) {
         layerExportsMap: Immutable.Map()
     });
 
+    Object.defineProperties(DocumentExports.prototype, objUtil.cachedGetSpecs({
+        /**
+         * Produce an array of ExportAssets for the root of this document, converted to a JS object
+         * This is used for serializing the data to Ps metadata store
+         *
+         * @return {Array.<object>} An array of ExportAsset-like JS objects
+         */
+        rootExportsArray: function () {
+            if (this.rootExports && this.rootExports.size > 0) {
+                return this.rootExports.toJS();
+            } else {
+                return [];
+            }
+        },
+
+        /**
+         * Get a list of layer IDs that have a non-empty set of configured export assets
+         *
+         * @return {Immutable.IndexedSeq.<number>}
+         */
+        layerIDsWithExports: function () {
```

Comment:
All I want is set of the layer IDs, which in this case come from the *key* of the map I'm operating on. So, I used `.keySeq()` to get just the keys (after filtering).  If that seems wrong to you, another obvious solution would be for this function to just return the filtered `map`, and have the consumer test with `map.has()` instead of `seq.includes()` like it is now.